lib/nx_stencil.ex

```diff
@@ -70,40 +84,65 @@ defmodule NxStencil do
     worker(parent, rank, up, down, top_halo, data, bottom_halo, iters - 1)
   end
 
-  def process_start do
+  defp process_start do
     receive do
-      {parent, :start_data,
-       {
-         rank,
-         up,
-         down,
-         top_halo,
-         init_data,
-         bottom_halo,
-         iters
-       }} ->
+      %InitData{
+        parent: parent,
+        data: %StencilData{
+          up_rank: up,
+          self_rank: rank,
+          down_rank: down,
+          top_halo: top_halo,
+          self_data: init_data,
+          bottom_halo: bottom_halo
+        },
+        iters: iters
+      } ->
         worker(parent, rank, up, down, top_halo, init_data, bottom_halo, iters)
     end
   end
 
-  def get_data(x_dim, y_dim) do
+  defp all_not_nil(lst) do
+    lst
+    |> Stream.map(fn x -> x != nil end)
+    |> Enum.reduce(true, fn x, acc -> x and acc end)
```

Comment:
The list is already in memory and using `Stream` here won't gain anything. In fact, it's probably a little slower. This is also a great place to use `Enum.all?/2`

```suggestion
    Enum.all?(list, &(not is_nil(&1)))
```